src/functions/rolePolicyFactory.ts

```diff
@@ -0,0 +1,84 @@
+import { APIGatewayAuthorizerResult, Statement } from "aws-lambda";
+import { AccessHttpVerbMap } from "../models/AccessHttpVerbMap";
+import { ILogEvent } from "../models/ILogEvent";
+import { HttpVerb } from "../services/http-verbs";
+import Role, { getLegacyRoles } from "../services/roles";
+import StatementBuilder from "../services/StatementBuilder";
+import newPolicyDocument from "./newPolicyDocument";
+
+type NonEmptyArray<T> = [T, ...T[]];
+
+const accessToHttpVerbs: AccessHttpVerbMap = {
+  read: ["GET", "HEAD"],
+  write: ["*"],
+  view: ["GET"],
```

Comment:
If this is the legacy factory then we should be able to remove `view: ["GET"]`, the legacy roles strictly used `read` and `write`